ambry-clustermap/src/main/java/com/github/ambry/clustermap/ClusterMapUtils.java

```diff
@@ -528,6 +529,15 @@ static class PartitionSelectionHelper implements ClusterMapChangeListener {
       this.clusterManagerQueryHelper = clusterManagerQueryHelper;
       this.defaultPartitionClass = defaultPartitionClass;
       updatePartitions(clusterManagerQueryHelper.getPartitions(), localDatacenterName);
+      logger.debug("Number of partitions in data center {} {}", localDatacenterName, allPartitions.size());
+      for (String partitionClass : partitionIdsByClassAndLocalReplicaCount.keySet()) {
+        logger.debug("Partition class {}, partitions {}", partitionClass,
+            partitionIdsByClassAndLocalReplicaCount.get(partitionClass).values());
```

Comment:
It's better to use 'entrySet' instead of 'keySet' here for both readability and performance: 

```
Output of above program (in milliseconds) :

Using entrySet() in for-each loop : 50
Using keySet() in for-each loop : 76
Using entrySet() and iterator : 50
Using keySet() and iterator : 75
```

https://howtodoinjava.com/java/collections/hashmap/performance-comparison-of-different-ways-to-iterate-over-hashmap/